src/components/Categories.tsx

```diff
@@ -1,43 +1,31 @@
-import React, { useEffect, useState } from 'react'
+import React, { useState } from 'react'
 import './../css/categories.css'
 import { Link } from 'react-router-dom'
 import ky from 'ky'
+import { useQuery } from 'react-query'
 
 function Categories() {
-  const [categories, setCategories] = useState<Category[]>([])
   const [selectedCategory, setSelectedCategory] = useState('all')
 
-  const [products, setProducts] = useState<ProductItem[]>([])
+  const categoriesQuery = useQuery('categories', fetchCategories)
+  const productsQuery = useQuery('products', fetchProducts)
```

Comment:
Very nice! The cool thing about the hooks approach is how easy it is to abstract these into reusable custom hooks. 

It's incredibly common to need to call the same API endpoint in different components. Also moving your API calling logic into a hook hides implementation details from components.
```suggestion
  const categoriesQuery = useCategoriesQuery()
  const productsQuery = useProductsQuery()
```

These hooks can basically be a thin wrapper around the useQuery hook just with the QueryKey & async function defined.